projects/laji/test/+user/user.po.ts

```diff
@@ -0,0 +1,46 @@
+import { Page, expect } from '@playwright/test';
+
+export const login = async (page: Page) => {
+  const currentPath = new URL(page.url()).pathname;
+
+  // use PERSON_TOKEN based login if possible
+  if (process.env.PERSON_TOKEN) {
+    await page.goto(
+      `http://localhost:3000/user/login?token=${process.env.PERSON_TOKEN}&next=${currentPath}`
+    );
+    return;
+  }
+
+  // fallback to manual login
+
+  // there are three cases:
+  //  not yet in app -> navigate to the app
+  //  in the app -> click login link -> external login
+  //  in external login -> continue login process
+
+  // first navigate to the app on a fresh page instance
+  // eg. a fresh page instance has url "about:blank" in chromium
+  if (!page.url().match(/http.*/)) {
+    await page.goto('/');
+  }
+
+  // in the app -> navigate to external login
+  await page.waitForLoadState();
+  if (await page.locator('#login-link').isVisible()) {
+    await page.locator('#login-link').click(); // laji.fi login link
+  }
```

Comment:
The "god" function was necessary for person-token based login fallback since the person token based login was supposed to be used everywhere, but if we can't get that to work then yeah probably a good idea to split them... I'll split it to three exported functions, `login`, `lajiFiLogin` and `lajiAuthLogin`.

Reply:
I changed the logic that checks whether we are on lajifi or lajiauth to be url based and started using the fine grained `lajiAuthLogin` for cases where we expect the browser to be on laji-auth page... let me know if this fixes flakiness for you (this was not a source of flakiness for me) and if you prefer the approach.